ImagePipeline/ImagePipeline/Producers/FutureAccessListFetchProducer.cs

```diff
@@ -0,0 +1,63 @@
+﻿using FBCore.Concurrency;
+using ImagePipeline.Image;
+using ImagePipeline.Memory;
+using ImagePipeline.Request;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Windows.Storage.AccessCache;
+
+namespace ImagePipeline.Producers
+{
+    /// <summary>
+    /// Represents a future-access list fetch producer.
+    /// </summary>
+    public class FutureAccessListFetchProducer : LocalFetchProducer
+    {
+        internal const string PRODUCER_NAME = "FutureAccessListFetchProducer";
+
+        /// <summary>
+        /// Instantiates the <see cref="FutureAccessListFetchProducer"/>.
+        /// </summary>
+        public FutureAccessListFetchProducer(
+            IExecutorService executor,
+            IPooledByteBufferFactory pooledByteBufferFactory) : base(
+                executor,
+                pooledByteBufferFactory)
+        {
+        }
+
+        /// <summary>
+        /// Gets the encoded image.
+        /// </summary>
+        protected override Task<EncodedImage> GetEncodedImage(ImageRequest imageRequest)
+        {
+            string originalString = imageRequest.SourceUri.OriginalString;
+            string token = originalString.Substring(originalString.LastIndexOf(":") + 1);
```

Comment:
This needs to be URL-decoded: `urn:future-access-list:%7D123-455-789%7B`

Reply:
Fixed.